rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -766,6 +766,9 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
                 isNull('status')
                 ne('status', EXECUTION_SCHEDULED)
             }
+            and {
```

Comment:
`cleanupRunningJobsAsync`, and `findRunningExecutions` both do the same query, the code should probably be unified to build criteria builder in a single method.  Also, the findRunningExecutions query should be tested

Reply:
I've removed the change to that query. When I first added missed executions I didn't set the dateCompleted and so it would cause the cleanup to pick up the missed records and change them. 

As an aside I'm wondering if we could remove the duplication by changing the `cleanupRunningJobsAsync` to be:
```
def cleanupRunningJobsAsync(String serverUUID = null, String status = null, Date before = new Date()) {
    executorService.submit {
            cleanupRunningJobs(findRunningExecutions(serverUUID, before), status)
    }
}
```
Seems odd to look up the ids on the main thread then use the ids to lookup the objects in a different thread.

Reply:
yes that seems better